classlib/src/main/java/org/teavm/classlib/java/util/TSpliterators.java

```diff
@@ -0,0 +1,77 @@
+/*
+ *  Copyright 2017 Alexey Andreev.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.teavm.classlib.java.util;
+
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.function.IntConsumer;
+
+public class TSpliterators {
+
+    private TSpliterators() {
+    }
+
+    public static <T> TSpliterator<T> spliterator(Object[] array, int additionalCharacteristics) {
+        TArrayList<T> list = new TArrayList<T>();
+        for (Object element : array) {
+            list.add((T) element);
+        }
+
+        return list.spliterator();
+    }
+
+    public static <T> TSpliterator<T> spliterator(Collection<? extends T> c, int characteristics) {
+        return ((TCollection) c).spliterator();
+    }
+
+    public static TSpliterator.OfInt spliterator(int[] array, int additionalCharacteristics) {
+        return spliterator(array, 0, array.length, additionalCharacteristics);
+    }
+
+    public static TSpliterator.OfInt spliterator(int[] array, int fromIndex, int toIndex,
+                                                 int additionalCharacteristics) {
+        return new TSpliterator.OfInt() {
+            int index = fromIndex;
+
+            @Override
+            public TSpliterator.OfInt trySplit() {
+                return null;
+            }
+
+            @Override
+            public boolean tryAdvance(IntConsumer action) {
+                index++;
```

Comment:
Does this work properly. Can you add tests for these methods?

Reply:
Added unit tests. And you were indeed correct they were needed because the initial version contained a mistake.